Tasks/JavaToolInstallerV0/FileExtractor/JavaFilesExtractor.ts

```diff
@@ -62,13 +62,33 @@ export class JavaFilesExtractor {
         const sevenZip = taskLib.tool(this.getSevenZipLocation());
         sevenZip.arg('x');
         sevenZip.arg('-o' + destinationFolder);
+        // skip if exists
+        sevenZip.arg('-aos');
         sevenZip.arg(file);
         const execResult = sevenZip.execSync();
         if (execResult.code != taskLib.TaskResult.Succeeded) {
             taskLib.debug('execResult: ' + JSON.stringify(execResult));
         }
     }
 
+    public static getFileEnding(file: string): string {
+        let fileEnding = '';
```

Comment:
@alexander-smolyakov the problem we saw here is that there's a validation for file ending - it differs from file extension... For example, for filename.tar.gz is considered to have file ending '.tar.gz', but '.gz' as extension. There's also could be the case when there's a dot in filename (like 'file.name.tar.gz'), so we can't simply extract substring after first dot...

Reply:
@anatolybolshakov Oh, I see now. I think it will be useful to add information about these cases in the function description.

Reply:
That function was just moved here from `javatoolinstaller.ts` to avoid the circular dependency

Reply:
Yes, I think it also makes sense to describe difference between file ending/file extension here - I asked @ekaterina-tatanova to update comments since she reworked this method also and added some js doc for it - https://github.com/microsoft/azure-pipelines-tasks/pull/13203#discussion_r450080434